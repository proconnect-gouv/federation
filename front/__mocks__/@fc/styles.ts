import { StylesProviderProps } from '@fc/styles';
import React, { ReactElement } from 'react';

export const useStylesQuery = jest.fn();

export const useStylesContext = jest.fn();

export const useStylesVariables = jest.fn(() => []);

export const StylesProvider = jest.fn(
  ({ children }: StylesProviderProps) => children as ReactElement,
);

export const StylesContext = React.createContext(expect.any(Object));

export enum Strings {
  DOUBLE_DASH = '--',
}
