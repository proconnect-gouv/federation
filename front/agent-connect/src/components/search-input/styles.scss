@import './../../styles/_colors.scss';
@import './../../styles/_variables.scss';

#identity-provider-search #fi-search-form {
  label[for='fi-search-term'] {
    color: $charcoal-grey;
    display: block;
    font-size: 18px;
    height: auto;
    line-height: 1.8rem;
    padding-bottom: $spacer;
    text-align: center;
  }

  input[name='fi-search-term'] {
    border: 1px solid #d9d9d9;
    border-radius: 0.25rem;
    float: none;
    height: 60px;
  }

  button[type='submit'] {
    display: block;
    height: 60px;
    margin-left: 0;
    margin-top: $spacer;
    width: 100%;

    &:hover {
      text-decoration: underline;
    }
  }

  .ant-input-group {
    display: flex;
    flex-direction: column;
  }

  .ant-input-group-addon {
    background: transparent;
    display: block;
    flex: 1;
    width: 100%;
  }
}

@media (min-width: $breakpoint-mobile) {
  #identity-provider-search #fi-search-form {
    label[for='fi-search-term'] {
      text-align: center;
    }

    button[type='submit'] {
      margin-top: 0;
      width: 100%;
    }

    .ant-input-group {
      flex-direction: row;
    }

    .ant-input-group-addon {
      flex: 0;
      padding-left: $spacer;
      width: 210px;
    }
  }
}
