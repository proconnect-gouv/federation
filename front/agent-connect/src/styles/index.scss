// ----------------- VARIABLES
@import '_variables.scss';
@import '_colors.scss';

// ----------------- VENDORS
@import '~antd/dist/antd.css';
@import '~bootstrap/scss/bootstrap';
@import '~sass-rem';

// ----------------- CUSTOM
@import './helpers';
@import './vendors';
@import './fonts/montserrat';

*,
input,
textarea {
  font-family: 'Montserrat', sans-serif;
}

body,
html,
#root {
  // @include rem-baseline;

  color: $charcoal-grey;
  height: 100%;
  min-height: 100%;
}

a,
.btn-link {
  color: $charcoal-grey;

  &:hover {
    color: $primary;
  }
}

.unordered-list {
  list-style: none;
  margin: 0;
  padding: 0;

  li {
    display: flex;
    flex-direction: row;
    padding: 2px 5px;

    &::before {
      content: 'â€¢';
      font-size: 36px;
      line-height: 0.85;
      margin-right: 12px;
    }
  }
}
