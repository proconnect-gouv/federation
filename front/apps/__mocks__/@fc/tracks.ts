export const TracksListComponent = jest.fn();
