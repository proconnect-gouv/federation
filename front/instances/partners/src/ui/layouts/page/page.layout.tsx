import React from 'react';
import { Outlet } from 'react-router-dom';

export const PageLayout = React.memo(() => (
  <main className="fr-container fr-py-8v">
    <div className="fr-grid-row fr-grid-row--center">
      <Outlet />
    </div>
  </main>
));

PageLayout.displayName = 'PageLayout';
