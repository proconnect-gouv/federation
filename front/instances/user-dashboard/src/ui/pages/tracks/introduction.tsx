import React from 'react';

export const IntroductionComponent = React.memo(() => (
  <div>
    <h1 className="fr-h3 is-blue-france fr-mb-1w">Mon historique de connexion</h1>
    <p>
      Retrouvez toutes les connexions et échanges de données effectués via FranceConnect ces six
      derniers mois.
      <br />
      Cliquez sur une connexion pour en afficher les détails.
    </p>
  </div>
));

IntroductionComponent.displayName = 'IntroductionComponent';
