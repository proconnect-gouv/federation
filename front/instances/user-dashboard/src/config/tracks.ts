/* istanbul ignore file */

// declarative file
import { TracksConfig } from '@fc/tracks';

// API URLs
export const API_BASE_URL = `/api`;

export const Tracks: TracksConfig = {
  API_ROUTE_TRACKS: `${API_BASE_URL}/traces`,
  API_ROUTE_USER_INFOS: `${API_BASE_URL}/me`,
  LUXON_FORMAT_DATETIME_SHORT_FR: "D 'Ã ' T",
  LUXON_FORMAT_DAY: 'DDD',
  LUXON_FORMAT_HOUR_MINS: 'T',
  LUXON_FORMAT_MONTH_YEAR: 'LLLL yyyy',
  LUXON_FORMAT_TIMEZONE: 'z',
};
