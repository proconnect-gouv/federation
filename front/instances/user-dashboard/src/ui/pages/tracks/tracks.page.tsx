import classnames from 'classnames';
import React from 'react';
import { Helmet } from 'react-helmet';
import { useMediaQuery } from 'react-responsive';

import { TracksListComponent } from '@fc/tracks';

import { AppConfig } from '../../../config';
import { IntroductionComponent } from './introduction';

export const TracksPage = React.memo(() => {
  const gtTablet = useMediaQuery({ query: '(min-width: 992px)' });
  return (
    <React.Fragment>
      <Helmet>
        <title>Mon tableau de bord - Historique</title>
      </Helmet>
      <div
        className={classnames('fr-m-auto fr-px-2w', {
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fr-mt-5w': !gtTablet,
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fr-mt-8w': gtTablet,
        })}
        id="page-container">
        <IntroductionComponent />
        <TracksListComponent options={AppConfig.Tracks} />
      </div>
    </React.Fragment>
  );
});

TracksPage.displayName = 'TracesPageComponent';
