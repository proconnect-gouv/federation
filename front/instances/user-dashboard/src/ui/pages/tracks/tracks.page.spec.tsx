import { render } from '@testing-library/react';

import { useStylesQuery, useStylesVariables } from '@fc/styles';
import { TracksListComponent } from '@fc/tracks';

import { IntroductionComponent } from './introduction';
import { TracksPage } from './tracks.page';

jest.mock('./introduction');

describe('TracksPage', () => {
  const TracksListComponentMock = jest.mocked(TracksListComponent);
  TracksListComponentMock.mockReturnValue(<div>FooBar TracksListComponent</div>);

  beforeEach(() => {
    // Given
    jest.mocked(useStylesVariables).mockReturnValue([expect.any(String)]);
  });

  it('should match the snapshot', () => {
    // Given
    jest.mocked(useStylesQuery).mockReturnValue(true);

    // When
    const { container } = render(<TracksPage />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should call IntroductionComponent', () => {
    // When
    render(<TracksPage />);

    // Then
    expect(IntroductionComponent).toHaveBeenCalled();
  });

  it('should called TracksListComponent', () => {
    // When
    render(<TracksPage />);

    // Then
    expect(TracksListComponent).toHaveBeenCalled();
  });
});
