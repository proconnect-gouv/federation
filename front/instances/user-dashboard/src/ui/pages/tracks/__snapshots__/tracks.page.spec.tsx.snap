// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TracksPage should match the snapshot 1`] = `
<div>
  <div
    data-mockid="Helmet"
  >
    <div>
      React Helmet Async
    </div>
  </div>
  <div
    class="fr-m-auto fr-px-2w fr-mt-8w"
    id="page-container"
  >
    <div>
      FooBar IntroductionComponent
    </div>
    <div>
      FooBar TracksListComponent
    </div>
  </div>
</div>
`;
